core/workspace_comment_render_svg.js

```diff
@@ -0,0 +1,459 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Methods for rendering a workspace comment as SVG
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+goog.provide('Blockly.WorkspaceCommentSvg.render');
+
+goog.require('Blockly.WorkspaceCommentSvg');
+
+
+/**
+ * Size of the resize icon.
+ * @type {number}
+ * @const
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.RESIZE_SIZE = 8;
+
+/**
+ * Radius of the border around the comment.
+ * @type {number}
+ * @const
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.BORDER_RADIUS = 3;
+
+/**
+ * Offset from the foreignobject edge to the textarea edge.
+ * @type {number}
+ * @const
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.TEXTAREA_OFFSET = 2;
+
+/**
+ * Offset from the top to make room for a top bar.
+ * @type {number}
+ * @const
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.TOP_OFFSET = 10;
+
+/**
+ * Returns a bounding box describing the dimensions of this comment.
+ * @return {!{height: number, width: number}} Object with height and width
+ *    properties in workspace units.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.getHeightWidth = function() {
+  return { width: this.getWidth(), height: this.getHeight() };
+};
+
+/**
+ * Renders the workspace comment.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.render = function() {
+  if (this.rendered_) {
+    return;
+  }
+
+  var size = this.getHeightWidth();
+
+  // Add text area
+  // TODO: Does this need to happen every time?  Or are we orphaning foreign
```

Comment:
Is this still a real TODO? If so, can you file an issue to track it. Seems important not to leak.

Reply:
It's fine because of the if(this.rendered_) above.  I'll remove it.